### ISSUE TITLE
can't use LUCENE and NEAR in the same query

### ISSUE THREAD

**Issue by anegrin** [orientechnologies/orientdb#5264]
orientdb: 
```
  2.1.5
```
query: 
```SQL
select @rid from MyClass where name LUCENE 'test*' and  ([address.lat,address.lng,$spatial] NEAR [35.6129021,13.44560550000002,{"maxDistance": 5}])
```

MyClass has an embedded property "address" with lat and lng

error is:
```
OQueryParsingException: Error on parsing command at position #0: Incompatible conditions found: name LUCENE ? [address.lat, address.lng, $spatial] NEAR [?, ?, {"maxDistance": ?}] 
```

performing
```SQL
select @rid from MyClass where name LUCENE 'test*'
```
or 
```SQL
select @rid from MyClass where ([address.lat,address.lng,$spatial] NEAR [35.6129021,13.44560550000002,{"maxDistance": 5}])
```
works...

**Comment by maggiolo00** [comment]
@anegrin how did you define the spatial index?

**Comment by anegrin** [comment]
fulltext code:
```SQL
CREATE INDEX MyClass_name_FULLTEXT_LUCENE ON MyClass(name) FULLTEXT ENGINE LUCENE
```

spatial code:
```SQL
CREATE INDEX MyAddress_lat_lng_SPATIAL_LUCENE ON MyAddress(lat,lng) SPATIAL ENGINE LUCENE
```

where MyClass.address type is MyAddress

**Comment by maggiolo00** [comment]
hi @anegrin 

spatial index is not supported as embedded.

You should save records directly in MyAddress class in order to use spatial index.
For multiple lucene we will remove the static check but only 1 index will be used.
That means 

```
select @rid from MyClass where name LUCENE 'test*' and  ([address.lat,address.lng,$spatial] NEAR [35.6129021,13.44560550000002,{"maxDistance": 5}])
```
The second condition will be evaluated in memory.

**Comment by anegrin** [comment]
thank you! i will just denormalize with a copy of lat and lng in the MyClass entity!

thanx again

**Comment by anegrin** [comment]
well i just moved lat and lng in MyClass and i get the same error. maybe i misunderstood your message...

**Comment by maggiolo00** [comment]
@anegrin  

there is indeed now the static check for multiple lucene condition. 
We will remove it in the next release

**Comment by maggiolo00** [comment]
@anegrin 
This should be fixed in 2.2.x. Just check in 2.2.x and worked without problem.

Thanks
